decidim-templates/lib/decidim/templates/admin_engine.rb

```diff
@@ -57,12 +57,6 @@ class AdminEngine < ::Rails::Engine
         end
       end
 
-      initializer "decidim_templates_admin.mount_routes" do
-        Decidim::Core::Engine.routes do
-          mount Decidim::Templates::AdminEngine, at: "/admin/templates", as: "decidim_admin_templates"
```

Comment:
Why don't we use something like: 
```

      initializer "decidim_templates_admin.mount_routes" do
        Decidim::Admin::Engine.routes do
          mount Decidim::Templates::AdminEngine, at: "/templates", as: "decidim_admin_templates"
        end
      end
```

Reply:
I agree that introducing coupling here is a step back. 

Sadly, what you're proposing doesn't work, as it gives the behaviour that started this whole PR.

```ruby
$ bin/rspec decidim-admin/spec/system/admin_access_spec.rb

  1) AdminAccess when the user is a normal user behaves like a 404 page visiting the page leads to a 404
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }
       expected "/admin/" to equal "/admin/templates/questionnaire_templates"
     Shared Example Group: "a 404 page" called from ./spec/system/admin_access_spec.rb:29
     # /home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/core/test/shared_examples/errors.rb:22:in `block (3 levels) in <top (required)>'

(...)

Finished in 5.9 seconds (files took 3.41 seconds to load)
7 examples, 1 failure

Failed examples:

rspec ./spec/system/admin_access_spec.rb:29 # AdminAccess when the user is a normal user behaves like a 404 page visiting the page leads to a 404

(...)
``` 

For respecting the `decidim-admin` module behaviour, we need to mount in their routes. Other modules don't have this problem because they're being mount by using a Registry/Manifest. For instance:

Spaces:

https://github.com/decidim/decidim/blob/a44de3d3c6bb6a6fd58dae7eae2a70f4b2ca731a/decidim-admin/config/routes.rb#L18-L20

Authorizations:

https://github.com/decidim/decidim/blob/a44de3d3c6bb6a6fd58dae7eae2a70f4b2ca731a/decidim-admin/config/routes.rb#L38-L40

As Templates isn't neither a Space nor an Authorization, then it doesn't implement any manifest that we can mount. 

What we could/should do if we don't want to introduce this coupling is to create a new Registry Manifest for a more generic kind of section in the Admin panel. But doing that would probably mean that this can't be a `type: fix` any more, so that's out of the scope of the current PR (IMO).

Reply:
Ok, let's mount it as it is, as we already have the dependency introduced by #10176
https://github.com/decidim/decidim/blob/61c0ff710b400fb7fed5242716077f3d63320112/decidim-admin/app/controllers/decidim/admin/application_controller.rb#L35-L37